source/virtualBuffers/gecko_ia2.py

```diff
@@ -23,9 +23,24 @@
 class Gecko_ia2_TextInfo(VirtualBufferTextInfo):
 
 	def _normalizeControlField(self,attrs):
+		#log.info("normalizeControlField attrs: %s", attrs)
 		ariaCurrent = attrs.get("IAccessible2::attribute_current")
 		if ariaCurrent != None:
 			attrs['current']= ariaCurrent
+		ariaPlaceholder = attrs.get("IAccessible2::attribute_placeholder")
+		# For efficiency, only check if it is valid to include placeholder when we have a placeholder value to include.
+		if ariaPlaceholder != None:
+			try:
+				start, end = self._getOffsetsFromFieldIdentifier(
+					int(attrs.get('controlIdentifier_docHandle')),
+					int(attrs.get('controlIdentifier_ID')))
+			except (LookupError, ValueError):
+				pass
+			else:
+				# only allow placeholder attribute if the field value is empty
+				controlFieldText = self.obj.makeTextInfo(textInfos.offsets.Offsets(start, end)).text

```

Comment:
Since you have the offsets, you can avoid potentially fetching a huge amount of text by checking if the offsets span only one character. Even an empty field will render a space in browse mode, so you'll still need to check for `" "`.